### PR TITLE
Add more request types to MemMsg

### PR DESCRIPTION
**Pull request by qtt2**
This PR adds more request types needed for LR, SC instructions and cache invalidation/flush. This also adds support for the extra types and a better line trace in MemoryCL.

### CODE REVIEW

File: pymtl3/stdlib/cl/MemoryCL.py [hunk 1]
```diff
@@ -89,26 +94,60 @@ def up_mem():
...
+
+          # Invalid type
+          else:
+            assert( False )
+
           s.resp_qs[i].enq( resp )
 
+          # update trace
+          s.trace += f"[{i}] {MemMsgType.str[req.type_.uint()]} "
+          s.trace += f"0x{req.addr.uint():0>8x} "
```
Comment (jsn1993) [inline pymtl3/stdlib/cl/MemoryCL.py:144]: Can you implement the trance in line_trace()?

Comment (qtt2) [reply pymtl3/stdlib/cl/MemoryCL.py]: done

Comment (qtt2) [reply pymtl3/stdlib/cl/MemoryCL.py]: any other comment before we can merge this?

**Comment by qtt2** [comment]
Can I merge this PR?

**Pull request merged**